product_pricelist_direct_print/views/report_product_pricelist.xml

```diff
@@ -0,0 +1,100 @@
+        <t t-call="report.internal_layout">
+            <div class="page">
+                <h2>Price List</h2>
+
+                <div class="row mt32 mb32">
+                    <div class="col-xs-7">
+                        <strong>Price List Name</strong>:<br/>
+                        <span t-field="o.pricelist_id"/>
+                    </div>
+                    <div class="col-xs-2">
+                        <strong>Currency</strong>:<br/>
+                        <span t-field="o.pricelist_id.currency_id"/>
+                    </div>
+                    <div class="col-xs-3">
+                        <strong>Print date</strong>:<br/>
+                        <t t-esc="time.strftime('%Y-%m-%d')"/>
+                    </div>
+                </div>
+
+                <table class="table table-condensed">
+                    <thead>
+                        <tr>
+                            <th>
+                                <strong>Description</strong>
+                            </th>
+                            <th t-if="o.show_standard_price" class="text-right">
+                                <strong>Cost Price</strong>
+                            </th>
+                            <th t-if="o.show_sale_price" class="text-right">
+                                <strong>Sale Price</strong>
+                            </th>
+                            <th class="text-right">
+                                <strong>List Price</strong>
+                            </th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <t t-if="not o.show_variants">
+                            <t t-set="products" t-value="o.product_tmpl_ids"/>
+                        </t>
+                        <t t-if="o.show_variants">
+                            <t t-set="products"
+                               t-value="o.product_ids or o.product_tmpl_ids.mapped('product_variant_ids')"/>
+                        </t>
+                        <t t-foreach="(o.categ_ids or products.mapped('categ_id') or o.categ_ids.search([])).sorted(key=lambda x:x.name)" t-as="category">
+                            <tr>
+                                <td colspan="100">
+                                    <strong t-field="category.name"/>
+                                </td>
+                            </tr>
+                            <tr t-foreach="products and products.with_context(categ_id=category.id).filtered(lambda x: x.categ_id.id == x.env.context['categ_id']) or products.search([('categ_id','=',category.id)])"
+                                t-as="product">
+                                <td>
+                                    <t t-if="product.default_code">
+                                        [<span t-field="product.default_code"/>]
+                                    </t>
+                                    <span t-esc="product.name"/>
```

Comment:
Better to use `t-field='product.display_name'` to get product name with attribute values, or you will only get the template name.